src/SourceBrowser/src/BinLogToSln/Program.cs

```diff
@@ -14,8 +14,115 @@
+                if (sourceFiles.HasValue)
+                {
+                    int totalSourceFiles = sourceFiles.Value.Length;
+                    int notSupportedFiles = sourceFiles.Value.Count(sf => sf.Path.Contains(".notsupported.cs"));
+                    
+                    if (totalSourceFiles > 0)
+                    {
+                        // Higher score for projects with fewer notsupported files relative to total
+                        score += (totalSourceFiles - notSupportedFiles) * 100;
+                        
+                        // Bonus for having no notsupported files at all
+                        if (notSupportedFiles == 0)
+                        {
+                            score += 1000;
+                        }
+                    }
+                }
+
+                // Prefer more specific target frameworks
+                // This is a heuristic - longer framework names are typically more specific
+                string targetFramework = GetTargetFrameworkFromCommandLine(invocation.CommandLineArguments);
+                if (!string.IsNullOrEmpty(targetFramework))
+                {
+                    score += targetFramework.Length * 10;
+                    
+                    // Prefer platform-specific frameworks over generic ones
+                    if (targetFramework.Contains("linux") || targetFramework.Contains("windows") || targetFramework.Contains("osx"))
+                    {
+                        score += 500;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error calculating score for {invocation.ProjectFilePath}: {ex.Message}");
+                // Return a base score so we don't exclude this invocation entirely
+                score = 1;
+            }
+
+            return score;
+        }
+
+        private static string GetTargetFrameworkFromCommandLine(string commandLineArguments)
+        {
+            if (string.IsNullOrEmpty(commandLineArguments))
+                return null;
+
+            // Look for /p:TargetFramework=xxx pattern
+            var match = System.Text.RegularExpressions.Regex.Match(
+                commandLineArguments, 
+                @"/p:TargetFramework=([^\s;]+)", 
+                System.Text.RegularExpressions.RegexOptions.IgnoreCase);
+            
+            if (match.Success)
+            {
+                return match.Groups[1].Value;
+            }
+
+            // Look for --target-framework xxx pattern  
+            match = System.Text.RegularExpressions.Regex.Match(
+                commandLineArguments, 
+                @"--target-framework\s+([^\s]+)", 
+                System.Text.RegularExpressions.RegexOptions.IgnoreCase);
+            
+            if (match.Success)
+            {
+                return match.Groups[1].Value;
+            }
+
+            return null;
+        }

```

Comment:
This won't work.  The TargetFramework will not be present on the compiler's commandline.  Instead you need to modify the library that analyzes the binlog.  BinLogReader should be updated to capture the value of the TargetFramework property from the project evaluation that contains the compiler invocation instance.  This can be added to the CompilerInvocation type.

Fetching project properties is a good general purpose protocol for grabbing information about these and we'd like to use it for other hooks.